docs: update dashboard-variable docs (#88824)
grafana/grafana: docs/sources/dashboards/variables/add-template-variables/index.md

   - **On Dashboard Load:** Queries the data source every time the dashboard loads. This slows down dashboard loading, because the variable query needs to be completed before dashboard can be initialized.
   - **On Time Range Change:** Queries the data source every time the dashboard loads and when the dashboard time range changes. Use this option if your variable options query contains a time range filter or is dependent on the dashboard time range.
1. In the **Query** field, enter a query.

   - The query field varies according to your data source. Some data sources have custom query editors.

>    - Each data source defines how the variable values are extracted. The common implementation is to use every string value returned from the data source response as a variable value. Make sure to double-check the data source documentation.

Reviewer: ```suggestion
   - Each data source defines how the variable values are extracted. The typical implementation uses every string value returned from the data source response as a variable value. Make sure to double-check the documentation for the data source.
```

>    - Some data sources allow providing custom "display names" for the values. For instance, the PostgreSQL, MySQL, and Microsoft SQL Server plugins handle this by looking for fields named `__text` and `__value` in the result. Other data sources may look for `text` and `value` or use a different approach. Always remember to double-check the data source documentation.

Reviewer: ```suggestion
   - Some data sources let you provide custom "display names" for the values. For instance, the PostgreSQL, MySQL, and Microsoft SQL Server plugins handle this by looking for fields named `__text` and `__value` in the result. Other data sources may look for `text` and `value` or use a different approach. Always remember to double-check the documentation for the data source.
```

   - **On Time Range Change:** Queries the data source every time the dashboard loads and when the dashboard time range changes. Use this option if your variable options query contains a time range filter or is dependent on the dashboard time range.
1. In the **Query** field, enter a query.
   - The query field varies according to your data source. Some data sources have custom query editors.

>    - The exact way how the query variable data is extracted from the query results is specific to the given data source, please check the data source's documentation, but usually every string value returned from the data source is used as a possible variable value.
>      - Additionally, some data sources allow providing custom "display names" for the values, for example the PostgreSQL, MySQL and the Microsoft SQL Server plugins handle this by looking for fields named `__text` and `__value` in the result. Other data sources look for `text` and `value` or use a different approach. Again, please refer to the documentation for the specific data source plugin.

Reviewer 2: 💡 Maybe something like:

```suggestion
- Each data source defines how the variable values are extracted. The common implementation is to use every string value returned from the data source response as a variable value. Make sure to double-check the data source documentation.
- Some data sources allow providing custom "display names" for the values. For instance, the PostgreSQL, MySQL, and Microsoft SQL Server plugins handle this by looking for fields named __text and __value in the result. Other data sources may look for `text` and `value` or use a different approach. Always remember to double-check the data source documentation.
```
